pkg/servicedisc/autoconnect.go

```diff
@@ -0,0 +1,99 @@
+package servicedisc
+
+import (
+	"context"
+	"time"
+
+	"github.com/skycoin/dmsg/cipher"
+	"github.com/skycoin/skycoin/src/util/logging"
+
+	"github.com/skycoin/skywire/internal/netutil"
+)
+
+const (
+	// PublicServiceDelay defines a delay before adding transports to public services.
+	PublicServiceDelay = 5 * time.Second
+
+	fetchServicesDelay = 2 * time.Second
+)
+
+// ConnectFn provides a way to connect to remote service
+type ConnectFn func(context.Context, cipher.PubKey) error
+
+// CheckConnFN checks that connection is alive
+type CheckConnFN func(cipher.PubKey) bool
+
+// Autoconnector continuously tries to connect to services
+type Autoconnector interface {
+	Run(context.Context, ConnectFn, CheckConnFN) error
+}
+
+type autoconnector struct {
+	client   *HTTPClient
+	maxConns int
+	log      *logging.Logger
+	conns    map[cipher.PubKey]struct{}
```

Comment:
I guess we are currently keeping a list of conns here to be able to differentiate between autoconnect transports and manually added user transports. If thats the only case to keep them here, then this PR might profit from merging #686 being merged first. I think we could just use the transport labels to differentiate between the autoconnect tps instead of keeping a separate structure for this purpose.